template/packages/backend/src/config.service.ts

```diff
@@ -12,10 +13,15 @@ export class ConfigService {
   constructor() {
     config()
     try {
+      // using @next/env to mirror frontend env handling.
+      // https://github.com/vercel/next.js/blob/7e387f0fce09b77f1cf2a777e2b5fbed874fbec0/docs/basic-features/environment-variables.md#loading-environment-variables
+      const env = loadEnvConfig("./", process.env.NODE_ENV !== "production")
```

Comment:
@hallettj Is there a better check we can do here?

Reply:
I think this is the right idea. ~~But we can do `process.env.NODE_ENV === "development" to match the `isDev` method further down. I think the dev script explicitly sets that variable.~~ Nope, we're actually setting `NODE_ENV` in `.env`. On reflection we shouldn't do that because it gets us into some circular logic.

The development mode check should be true if `NODE_ENV` is neither `production` nor `test`. The simplest option is to import [`isdev`](https://www.npmjs.com/package/isdev). Can you either do that, or write a new helper function, and use it for both this check, and for the `isDev` method? Our `isDev` method should read from `process.env`, not from `this.env`. And we should remove `NODE_ENV` from the `.env` files.